### PR TITLE
Modernize plugin, drop pipeline support (JEP-210)

### PR DESCRIPTION
**Pull request by daniel-beck**

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
JEP-210 was delivered in Pipeline plugins, but there are only a few reference implementations. I am not sure its existence justifies the feature removal if it works in non-JEP-210 scenarios

**Comment by oleg-nenashev** [comment]
CC @jglick

**Review by TobiX — APPROVED** [review]
This is probably the best way forward for this plugin.

**Comment by daniel-beck** [comment]
@oleg-nenashev 

> if it works in non-JEP-210 scenarios

What scenarios are those? People who haven't updated their pipeline plugins in years?

**Review by oleg-nenashev — APPROVED** [review]
Fair point. I agree it is better to move forward here. A major version release might make sense

File: pom.xml [hunk 1]
```diff
@@ -14,13 +14,14 @@
   <url>https://wiki.jenkins-ci.org/display/JENKINS/Compress+Build+Log+Plugin</url>
```
Comment (jglick) [inline pom.xml]:
```suggestion
  <url>https://github.com/jenkinsci/compress-buildlog-plugin</url>
```

?

File: src/main/java/org/jenkinsci/plugins/compressbuildlog/BuildLogCompressor.java [hunk 2]
```diff
@@ -99,15 +105,7 @@ public void onFinalized(Run run) {
...
-                // regular, expected log file
-                FileInputStream fis = null;
-                FileOutputStream fos = null;
-                GZIPOutputStream gzos = null;
-
-                try {
-                    fis = new FileInputStream(log);
-                    fos = new FileOutputStream(new File(log.getParentFile(), gzippedLogName));
-                    gzos = new GZIPOutputStream(fos);
+                try(FileInputStream fis = new FileInputStream(log); GZIPOutputStream gzos = new GZIPOutputStream(new FileOutputStream(new File(log.getParentFile(), gzippedLogName)))) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/compressbuildlog/BuildLogCompressor.java]:
```suggestion
                try(FileInputStream fis = new FileInputStream(log); FileOutputStream fos = new FileOutputStream(new File(log.getParentFile(), gzippedLogName)); GZIPOutputStream gzos = new GZIPOutputStream(fos)) {
```

to ensure `fos` gets closed

File: src/test/java/org/jenkinsci/plugins/compressbuildlog/BuildLogCompressorPipelineTest.java [hunk 3]
```diff
@@ -20,6 +21,6 @@ public void testIsApplicable() throws Exception {
         Assert.assertNotNull(descriptor);
 
         boolean res = descriptor.isApplicable(workflowJob.getClass());
-        Assert.assertTrue(res);
+        Assert.assertFalse(res);
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/compressbuildlog/BuildLogCompressorPipelineTest.java]:
```suggestion
        Assert.assertFalse("TODO does not work with post-JEP-210 Pipeline", res);
```

File: README.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+* https://issues.jenkins-ci.org/browse/JENKINS-24171[JENKINS-24171]
+
+== Changelog
+
+=== Upcoming Changes
+
+* *Require Jenkins 2.150.1*
+* General modernization
+* Drop support for Pipelines due to JEP-210
```
Comment (jglick) [inline README.adoc]:
```suggestion
* Drop support for Pipelines due to link:https://jenkins.io/jep/210[JEP-210]
```

**Review by jglick — APPROVED** [review]
Fine with me but I would suggest editing the PR title to clarify that you are dropping support for _obsolete_ Pipeline versions. Adding support for current Pipeline versions would likely not be very difficult but would mean providing a JEP-210 implementation, as I think is covered in some existing JIRA RFE.

**Comment by hcldan** [comment]
Please do not drop pipeline support...   I realize that there are problems... but it would be nice if they were fixed instead of just removing support.

**Comment by daniel-beck** [comment]
This change doesn't mean that there never will be support for JEP-210 Pipelines, just that the current state is a problem.

This has been broken for roughly 1.5 years. If you've managed to not update your Pipeline plugins in that time, you can manage to not update this one as well.

Otherwise, this has been broken and nobody cared enough to repair it. So I'm bringing the plugin back to a sane state. If anyone wants to add JEP-210 support, nothing's stopping them. I'd be happy to review and integrate PRs that do that.

**Review by TobiX — APPROVED** [review]

**Comment by mamh2021** [comment]
what about this PR :https://github.com/jenkinsci/workflow-api-plugin/pull/95/

**Comment by mchoma** [comment]
Hi all. What is plan to merge this PR and release plugin with this fix? I see there are already 3 approves of this PR.

According to my information issue this PR is fixing is preventing us from enabling this plugin for us.

**Comment by daniel-beck** [comment]
I don't remember why I self-assigned this, and I also am no longer a maintainer of this plugin, only Oleg is. It's probably best if I close this PR and let others with active interest in making this work move the plugin forward. Sorry everyone.

**Pull request closed without merging**